### DISCUSSION TITLE
jupyter-book and textual formats like py:percent

### DISCUSSION THREAD

**Discussion by parmentelat** [executablebooks/meta#273] [Q&A] [answered]
Hi

my understanding is that jupyter-book can be used on natively on notebooks stored as ipynb, or md:myst

turns out I have a lot of py:percent notebooks; of course, as the jupyter-book doc mentions, it's easy to use jupytext to translate py:percent into md:myst, but my feeling is that a smooth tooling should ideally be able to take care of that (in particular in order to keep an incremental workflow where only changed notebooks trigger a conversion + execution)

where is the team standing on this ? 
is it that it's actually supported and I failed to spot it ?
or is it that nobody has yet the time to tackle that ?
or is it intentionally left out b/c deemed as outside of the project scope ?

thanks for your kind feedback

**Comment by chrisjsewell** [comment, marked as answer]
Heya

> in particular in order to keep an incremental workflow where only changed notebooks trigger a conversion + execution


Yeh this should already work, have you tried it?

**Reply by chrisjsewell** [reply]
As in by using https://jupyterbook.org/file-types/jupytext.html

**Reply by parmentelat** [reply]
all righty - sorry for not spotting that earlier - thanks !!

**Reply by chrisjsewell** [reply]
no worries 😄

**Comment by parmentelat** [comment]
so just for the record, it appears that

* I need to have at least one `file` entry at the toplevel of the `_toc` (I would even assume, exactly one)
* and this one file, apparently again, must be in either ipynb, md or rst format; that's what was suggested by the error message that I was getting, when in my case that landing page was in `py:percent` format; the error vanished when I translated it manually into `md:myst' 

no big deal of course, was just a little surprising, given that all my notebooks were in py:percent

**Reply by chrisjsewell** [reply]
thanks for the feedback yes I seem to recall maybe someone else having this issue that @choldgraf was talking to

**Reply by choldgraf** [reply]
Yeah I think we need to improve the documentation around text based formats, and i think there were some assumptions in the code that we need to loosen. IIRC there's a section that checks the extension in the toc code and if it isn't a subset of allowed ones, displays a misleading error. I just haven't had the time to dig into it and figure it out
